### ISSUE TITLE
Падает Mahou при работе с исключениями

### ISSUE THREAD

**Issue by Makishima** [BladeMight/Mahou#314]
У меня довольно часто падает, если что-то изменить в программах исключениях и применить настройки. Иногда падает, если оттуда что-то удалить или вставить горячими клавишами.
Что-то кроме логов можно собрать?
[2021.01.14.txt](https://github.com/BladeMight/Mahou/files/5813713/2021.01.14.txt)

**Comment by BladeMight** [comment]
> Что-то кроме логов можно собрать?

Логи с Debug версий, [Mahou-latest-commit](https://github.com/BladeMight/Mahou/releases/tag/latest-commit) (бери Debug_x86_x64.zip)
И в твоих логах нет "падения", включи логи и сделай так чтобы попалось "падение".

**Comment by BladeMight** [comment]
> и применить настройки

Возможно из-за автозамены? Т.к. каждый раз когда нажимаешь применить "автозамена" переинициализируется. У тебя точно хватает памяти? Mahou с тем словарём после нескольких переинициализаций может брать до 1000+МБ.

**Comment by Makishima** [comment]
Падало и на маленьком словаре, посижу пока что на дебаге.

**Comment by Makishima** [comment]
При падении предполагается создание дампа в директории с Mahou или просто логи будут расширенные?

**Comment by BladeMight** [comment]
> и просто логи будут расширенные?

Вот это.

**Comment by Makishima** [comment]
А дебаг версия если обновится, то на обычную, да? :)

**Comment by BladeMight** [comment]
>  то на обычную, да?

Да. При обновлении скачивает Release_x86_x64.zip.

**Comment by Makishima** [comment]
Вот лог
[2021.01.17.txt.zip](https://github.com/BladeMight/Mahou/files/5826611/2021.01.17.txt.zip)

**Comment by BladeMight** [comment]
Спасибо за потраченное время, но в логе нет ничего такого что говорило бы о том что была критическая ошибка, возможно какое-то другое приложение закрывает Mahou? Или может ошибка связанная с "полным восстановлением буфера обмена", см https://github.com/BladeMight/Mahou/issues/305#issuecomment-756383199 и ниже, чтобы проверить вот последняя версия где "полного восстановления буфера обмена" нет, а есть только "восстановление текста буфера обмена":
https://ci.appveyor.com/project/BladeMight/mahou/builds/37108413/artifacts

**Comment by Makishima** [comment]
Вряд ли, здесь падение только при работе с окном программы возникает.

**Comment by BladeMight** [comment]
> Вряд ли

что именно: "полного восстановления буфера обмена" или "другое приложение"?

**Comment by Makishima** [comment]
Оба варианта маловероятны. Захожу в главное окно, меняю там настройки или список исключений и есть шанс, что программа падает

**Comment by BladeMight** [comment]
> Захожу в главное окно, меняю там настройки или список исключений и есть шанс, что программа падает

Но ведь когда изменяешь настройки - иногда можешь используешь буфер обмена? Для копирования, вырезания, вставки, или используя функции "конвертации выделения" Mahou которые также используют буфер обмена.

...И давно это происходит?

**Comment by Makishima** [comment]
Перед НГ вроде замечал и несколько раз после, поэтому и завел.
Может и раньше - первый лаз словил, когда нужно было расширить программы-исключения, и, да, в список вставлял имя процесса, скопировав его из ProcessExplorer

**Comment by BladeMight** [comment]
@Makishima Возможно это тоже связано, судя по тому что ты писал:

>в программах исключениях и __применить__ настройки

в 7e116b4, сделал чтобы словарь Автозамены, когда очень большой (Mahou напишет об этом в месте где обычно редактируешь словарь), не обновлялся каждый раз по кнопке "Применить", а только 1 раз при запуске или при выключении=>Применить=>включении=>Применить функции "Автозамены".

**Comment by Makishima** [comment]
Сейчас упал при открытии главного окна. К сожалению, логи не писал, так как давно падений не было

**Comment by BladeMight** [comment]
> Сейчас упал при открытии главного окна. К сожалению, логи не писал, так как давно падений не было

Я уже у себя тоже заметил, у меня были включены логи но всё равно ничего не записало, видимо что-то "конкретное" происходит раз из .Net не получается лог ошибки получить.

Мне всё же кажется что из-за:

> "полным восстановлением буфера обмена", см https://github.com/BladeMight/Mahou/issues/305#issuecomment-756383199 и ниже

Попробуй использовать эту сборку (она без "полного восстановления буфера обмена"):
[Mahou-NoClip-x86_64.zip](https://github.com/BladeMight/Mahou/files/5924198/Mahou-NoClip-x86_64.zip)

**Comment by Makishima** [comment]
А версию он как `fc31776` показывает - это ок?

**Comment by BladeMight** [comment]
Да, это из `Mahou.ini`, по "последнему" обновлению, т.к. в этой версии "нету" внедренной строки версии.
